x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/transport/TransportFindFieldStructureAction.java

```diff
@@ -0,0 +1,214 @@
+        Map<String, String> headers,
+        ActionListener<TextStructureResponse> listener
+    ) {
+
+        ActionListener<SearchResponse> searchResponseActionListener = ActionListener.wrap(searchResponse -> {
+            // TODO could we just feed the doc fields as "lines" to the field structure finder?
+            StringBuilder stringBuilder = new StringBuilder();
+            for (SearchHit hit : searchResponse.getHits().getHits()) {
+                if (request.getRuntimeMappings().isEmpty()) {
+                    Object value = MapHelper.dig(request.getFieldName(), Objects.requireNonNull(hit.getSourceAsMap()));
+                    if (value instanceof String) {
+                        stringBuilder.append(value.toString()).append("\n");
+                    } else if (value instanceof List<?>) {
+                        @SuppressWarnings("unchecked")
+                        List<Object> values = (List<Object>) value;
+                        for (Object v : values) {
+                            if (v instanceof String) {
+                                stringBuilder.append(v.toString()).append("\n");
+                            }
+                        }
+                    }
+                } else {
+                    DocumentField field = hit.field(request.getFieldName());
+                    for (Object v : field.getValues()) {
+                        if (v instanceof String) {
+                            stringBuilder.append(v.toString()).append("\n");
+                        }
+                    }
+                }
+            }
+            threadPool.executor(GENERIC).execute(() -> {
+                try {
+                    listener.onResponse(buildTextStructureResponse(request, new BytesArray(stringBuilder.toString())));
+                } catch (Exception e) {
+                    listener.onFailure(e);
+                }
+            });
+        }, listener::onFailure);
+        FunctionScoreQueryBuilder functionBuilder = request.getQueryBuilder() != null
+            ? QueryBuilders.functionScoreQuery(request.getQueryBuilder(), ScoreFunctionBuilders.randomFunction())
+            : QueryBuilders.functionScoreQuery(ScoreFunctionBuilders.randomFunction());
+
+        SearchRequestBuilder searchRequest = client.prepareSearch(request.getIndices())
+            .setQuery(functionBuilder)
+            .setSize(request.getLinesToSample())
+            .setTrackTotalHits(false);
+        // If we have a runtime mapping, that means we have to fetch via the fields
+        // But, don't do that unless we have to so that the user could query older remote clusters
+        if (request.getRuntimeMappings().isEmpty() == false) {
+            searchRequest.addFetchField(request.getFieldName()).setFetchSource(false).setRuntimeMappings(request.getRuntimeMappings());
+        } else {
+            searchRequest.setFetchSource(new String[] { request.getFieldName() }, null);
+        }
```

Comment:
There is also the broader consideration of "if we have remote indices, we should route to a remote_client + data node". But, didn't want to bother with that until we see a need for this API